tools/test/examples/examples_lib.py

```diff
@@ -399,28 +410,54 @@ def compile_repos(config, toolchains, targets, profile, verbose, examples):
                                                                 example['features']):
                     
                     build_command = ["mbed-cli", "compile", "-t", toolchain, "-m", target] + (['-vv'] if verbose else [])
-                    if profile:
-                        build_command.append("--profile")
-                        build_command.append(profile)
+                    if profiles:
+                        for profile in profiles:
+                            build_command.extend(["--profile", profile])
                     
                     print("Compiling [%s] for [%s] with toolchain [%s]\n\n>  %s" % (name, target, toolchain, " ".join(build_command)))
                     
-                    proc = subprocess.Popen(build_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                    if (
+                        post_checks
+                        and POST_BUILD_CHECK_FLOATS in post_checks
+                        and name not in POST_BUILD_CHECK_FLOATS_EXCLUDED_EXAMPLES
+                    ):
+                        # Disable floating point support for minimal-printf if
+                        # a post build check for minimal-printf is specified.
+                        has_float_support = "false"
+                    else:
+                        has_float_support = "true"
+
+                    _set_minimal_printf_floats_status(has_float_support)
```

Comment:
We should change the default library configuration only if we are checking for floats.
So I suggest
```
if (
   post_checks
   and POST_BUILD_CHECK_FLOATS in post_checks
    and name not in POST_BUILD_CHECK_FLOATS_EXCLUDED_EXAMPLES
  ):
   # Disable floating point support for minimal-printf if
   # a post build check for minimal-printf is specified.
 _set_minimal_printf_floats_status("false")
```

Reply:
https://github.com/ARMmbed/mbed-os/pull/11429#issuecomment-529945636